### PR TITLE
Snow context unware defaults and validations

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

#1105 

*Description of changes:*

Add context unaware set defaults and validations for snow provider with the new config manager.

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/snow/apibuilder_test.go [hunk 1]
```diff
@@ -283,6 +283,7 @@ func TestSnowMachineTemplate(t *testing.T) {
 	got := SnowMachineTemplate(tt.machineConfigs["test-wn"])
 	wantAMIID := "eks-d-v1-21-5-ubuntu-ami-02833ca9a8f29c2ea"
 	wantSSHKey := "default"
+	wantPhysicalNetworkConnector := "" // TODO: update this value once configManager is hooked
```
Comment (g-gaston) [inline pkg/providers/snow/apibuilder_test.go:286]: curious: why does this need to wait?

File: pkg/cluster/snow.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
+package cluster
+
+import anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+
+func snowEntry() *ConfigManagerEntry {
```
Comment (g-gaston) [inline pkg/cluster/snow.go:5]: We should probably add a test parsing one snow cluster
We don't need to reuse the same test as the other providers
It probably make sense to have one specific for snow that only checks snow datacenter and snow machines

**Review by g-gaston — COMMENTED** [review]
Two minor comments, rest looks good

File: pkg/api/v1alpha1/snowmachineconfig.go [hunk 3]
```diff
@@ -69,3 +72,35 @@ func GetSnowMachineConfigs(fileName string) (map[string]*SnowMachineConfig, erro
 	}
 	return configs, nil
 }
+
+func validateSnowMachineConfig(config *SnowMachineConfig) error {
```
Comment (g-gaston) [inline pkg/api/v1alpha1/snowmachineconfig.go:76]: We should test this, probably through `SnowMachineConfig.Validate()`

File: pkg/api/v1alpha1/snowmachineconfig.go [hunk 4]
```diff
@@ -69,3 +72,35 @@ func GetSnowMachineConfigs(fileName string) (map[string]*SnowMachineConfig, erro
...
+		return fmt.Errorf("SnowMachineConfig SshKeyName is a required field. Create one in Snow Console if not any")
+	}
+
+	if config.Spec.InstanceType != SbeCLarge && config.Spec.InstanceType != SbeCXLarge && config.Spec.InstanceType != SbeC2XLarge && config.Spec.InstanceType != SbeC4XLarge {
+		return fmt.Errorf("SnowMachineConfig InstanceType %s is not supported, please use one of the following: %s, %s, %s, %s ", config.Spec.InstanceType, SbeCLarge, SbeCXLarge, SbeC2XLarge, SbeC4XLarge)
+	}
+	return nil
+}
+
+func setSnowMachineConfigDefaults(config *SnowMachineConfig) {
```
Comment (g-gaston) [inline pkg/api/v1alpha1/snowmachineconfig.go:91]: We should test this, probably through `SnowMachineConfig.SetDefaults()`

File: pkg/cluster/snow.go (hunk 2 above)
Comment (jiayiwang7) [reply pkg/cluster/snow.go:5]: i will add one in existing parsing test including cluster that covers the Ref to snow objects

**Review by g-gaston — APPROVED** [review]
/lgtm

**Pull request merged**